### PR TITLE
[PT FE]: Add support for aten::replace

### LINKED ISSUE #29712: [Good First Issue]: Support aten::replace
### Context

As in the main issue.

### What needs to be done?

As in the main issue.

### Example Pull Requests

_No response_

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Intel DevHub Discord channel](https://discord.gg/7pVRxUwdWG) - engage in discussions, ask questions and talk to OpenVINO developers
- [How to link your Pull Request to an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#manually-linking-a-pull-request-to-an-issue-using-the-pull-request-sidebar)

### Contact points

As in the main issue.

### Ticket

_No response_

### PR DESCRIPTION
**Pull request by Ratish1**
### Details:
 -  Adds support for the aten::replace operation by creating a new C++ translator that maps it to the OpenVINO Select operation.
-  A new layer test is added to validate the functionality of the translator.

### Tickets:
 - Closes #29712

### CODE REVIEW

**Comment by Ratish1** [comment]
Hello @p-wysocki @PiotrKrzem @mmikolajcz , I wanted to follow up on this incasw you missed it. Thanks.

File: tests/layer_tests/pytorch_tests/test_replace.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        )
+
+    def create_model(self):
+        class aten_replace_model(torch.nn.Module):
+            def forward(self, x, target, replacement):
+                return torch.where(x == target, replacement, x)
+
+        ref_net = None
+
+        return aten_replace_model(), ref_net, "aten::where"
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_replace.py:25]: This test do not produce `aten::replace` operation

Comment (Ratish1) [reply tests/layer_tests/pytorch_tests/test_replace.py:25]: Hello @mvafin thanks for the feeback, using atten::where is wrong. I tested various PyTorch operations that could produce `aten::replace`:
- `x[mask] = value` produced `aten::index_put_`
- `x.masked_fill_(mask, value)` produced `aten::masked_fill_`
I couldnt find any documentation that `aten::replace` exists in pytorch operations.So could you guide me to how to produce a test for `aten::replace`

**Comment by Ratish1** [comment]
Hello @mvafin , I wanted to follow up on this in case you were busy. Thanks.
